hibernate-core/src/main/java/org/hibernate/persister/entity/AbstractEntityPersister.java

```diff
@@ -521,6 +521,8 @@ public boolean hasRowId() {
 		return propertySelectable;
 	}
 
+	public abstract String[] getTableNames();
```

Comment:
Not sure if `AbstractEntityPersister` can be considered an API/SPI but IMO a default implementation should just return null here to be safe.

Reply:
Thanks for the implementation of getTableNames(), I've done it like so.